src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Hover/DefaultRazorHoverInfoServiceTest.cs

```diff
@@ -492,6 +504,103 @@ public void GetHoverInfo_TagHelper_Attribute_VSClient_ReturnVSHover()
                 run => DefaultVSLSPTagHelperTooltipFactoryTest.AssertExpectedClassification(run, "BoolVal", VSPredefinedClassificationTypeNames.Identifier));
         }
 
+        [Fact]
+        public async Task Handle_Hover_SingleServer_CallsDelegatedLanguageServer()
+        {
+            // Arrange
+            var languageServerFeatureOptions = Mock.Of<LanguageServerFeatureOptions>(options => options.SupportsFileManipulation == true && options.SingleServerSupport == true, MockBehavior.Strict);
+
+            var delegatedHover = new VSInternalHover();
+            var responseRouterReturnsMock = new Mock<IResponseRouterReturns>(MockBehavior.Strict);
```

Comment:
Can be a follow up, but for one of my rename tests I manged to use the real Roslyn language server. Would be good to do that (and in fact, if done as a follow up, could be another opportunity for seeing if we can share more code)

Reply:
Added a test with the actual server